### PR TITLE
fix status handler conditions being always empty

### LINKED ISSUE #49: generated status handler processing never updates conditions
While working on the upgrade operator for k3os I've noticed that conditions are never updated and propagated even though the handler has mutated the status.

### PR DESCRIPTION
**Pull request by dweomer**
Fixes #49 that prevented status updates from accumulating conditions.

### CODE REVIEW

File: pkg/controller-gen/generators/type_go.go [hunk 1]
```diff
@@ -360,6 +360,7 @@ func (a *{{.lowerName}}StatusHandler) sync(key string, obj *{{.version}}.{{.type
 	}
 	if !equality.Semantic.DeepEqual(status, newStatus) {
 		var newErr error
+		newStatus.Conditions = obj.Status.Conditions
```
Comment (dweomer) [inline pkg/controller-gen/generators/type_go.go]: pickup the results of the `a.condition.SetError` operations from above

Comment (daxmc99) [reply pkg/controller-gen/generators/type_go.go]: Hmm, it looks likes newStatus never escapes this scope.

Comment (dweomer) [reply pkg/controller-gen/generators/type_go.go]: it's copied to `obj.Status` and then persisted via `UpdateStatus` on `obj`

**Comment by dweomer** [comment]
Superceded by #51

**Pull request closed without merging**